core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/Workflow.java

```diff
@@ -15,10 +15,21 @@
 
   List<String> getLog();
 
+  /**
+   * Validate and store input for the current step, which corresponds to the result of getNextStep.
+   */
   void processInput(ProgressStep step);
 
+  /**
+   * Validate and store input for a step identified by the 0-indexed stepNumber.
+   * stepNumber may refer to a previous step.
```

Comment:
Should mention that it is up to the implementation to invalidate later steps if necessary